.github/workflows/windows.yml

```diff
@@ -10,28 +10,35 @@ jobs:
     runs-on: ${{matrix.os}}
     strategy:
       matrix:
-        # https://github.com/actions/virtual-environments.
         os: [windows-2022]
-        platform: [x64]
-        toolset: [v142]
+        generator: ["Ninja", "Visual Studio 17 2022"]
         standard: [17, 20, 23]
-        build_type: [Debug]
+        # BUG: Ninja + MSVC + Modules fails to build
+        exclude:
+          - {generator: "Ninja", standard: 20 }
+          - {generator: "Ninja", standard: 23 }
 
     steps:
     - uses: actions/checkout@v3
+    - uses: ilammy/msvc-dev-cmd@v1
```

Comment:
What is this?

Reply:
For Ninja + MSVC you need to have `cl.exe` in the `PATH` environment variable. 

Usually people use `vcvarsall.bat` or `VsDevCmd.cmd` or specify the full path in `CMAKE_{C,CXX}_COMPILER` option. 

But for MSVS, the location of MSVC is not constant and can be changed with github workflows os image update.

To overcome this kind of problem, Microsoft came up with `vswhere.exe', which searches for the MSVC compiler.

It seems that `ilammy/msvc-dev-cmd` is the easiest way to use `vswhere` for github workflows.
It:
* finds `vswhere.exe` on the system
* runs it
* executes `vcvarsall.bat` for the last available MSVC, so we will have `cl.exe` in the `PATH` environment variable.

Reply:
MSVC + ninja is an unusual combination and considering this complexity I suggest only testing ninja on Linux.

Reply:
Rolling back. Perhaps Linux+Ninja will be enough, otherwise, we can come back to this later.

However, MSVC + Ninja is not unusual these days: it is the default for CLion+MSVC and Vcpkg, so the user base is actually quite large.